File: codegen/results/v002/m3-max-48gb/ollama/qwen2.5-coder_7b/20260411_224426/python/007_beatles_interview/solution.py
import csv
from datetime import datetime

# Define the transformation rules and output format based on expected_format.json
def transform_data(input_csv):
    output = []
    
    with open(input_csv, newline='') as csvfile:
        reader = csv.DictReader(csvfile)
        
        for row in reader:
            # Calculate age as of July 1, 2025
            birthdate = datetime.strptime(row['Birthday'], '%m/%d/%Y')
            current_date = datetime(2025, 7, 1)
            age = current_date.year - birthdate.year - ((current_date.month, current_date.day) < (birthdate.month, birthdate.day))
            
            # Prepare relatives data
            relatives = []
            if row['Father']:
                relatives.append({'FirstName': row['Father'], 'LastName': '', 'Relationship': 'Father'})
            if row['Mother']:
                relatives.append({'FirstName': row['Mother'], 'LastName': '', 'Relationship': 'Mother'})
            if row['Brother']:
                relatives.append({'FirstName': row['Brother'], 'LastName': '', 'Relationship': 'Brother'})
            if row['Sister']:
                relatives.append({'FirstName': row['Sister'], 'LastName': '', 'Relationship': 'Sister'})
            
            # Add the transformed data to the output list
            output.append({
                "FirstName": row['Name'].split()[0],
                "LastName": row['Name'].split()[1] if len(row['Name'].split()) > 1 else '',
                "Birthday": row['Birthday'],
                "Age": age,
                "Relatives": relatives
            })
    
    return output

# Read the input CSV and transform the data
output_json = transform_data('input/input.csv')

# Output the JSON array to stdout
import json
print(json.dumps(output_json, indent=2))

File: codegen/results/v002/m3-max-48gb/ollama/qwen2.5-coder_7b/20260411_224426/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

type Person struct {
	FirstName string `json:"FirstName"`
	LastName  string `json:"LastName"`
	Birthday  string `json:"Birthday"`
	Age       int    `json:"Age"`
	Relatives []Relative `json:"Relatives"`
}

type Relative struct {
	FirstName string `json:"FirstName"`
	LastName  string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

func main() {
	file, err := os.Open("input/input.csv")
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	reader := csv.NewReader(file)
	records, _ := reader.ReadAll()

	var people []Person

	for i, record := range records {
		if i == 0 {
			continue // Skip header row
		}

		nameParts := strings.Split(record[0], " ")
		firstName := nameParts[0]
		lastName := ""
		if len(nameParts) > 1 {
			lastName = nameParts[1]
		}

		birthday, _ := time.Parse("1/2/2006", record[1])
		age := calculateAge(birthday, 7, 1, 2025)

		var relatives []Relative
		addRelative(&relatives, "Father", record[3], lastName)
		addRelative(&relatives, "Mother", record[4], lastName)
		addRelative(&relatives, "Brother", record[5], lastName)
		addRelative(&relatives, "Sister", record[6], lastName)

		people = append(people, Person{
			FirstName: firstName,
			LastName:  lastName,
			Birthday:  birthday.Format("2006-01-02"),
			Age:       age,
			Relatives: relatives,
		})
	}

	jsonData, _ := json.MarshalIndent(people, "", "  ")
	fmt.Println(string(jsonData))
}

func calculateAge(birthday time.Time, month int, day int, year int) int {
	currentYear := year
	birthMonth := birthday.Month()
	birthDay := birthday.Day()

	if currentYear < birthday.Year() ||
		(currentYear == birthday.Year() && (birthMonth > month || (birthMonth == month && birthDay > day))) {
		currentYear--
	}

	return currentYear - birthday.Year()
}

func addRelative(relatives *[]Relative, relationship string, firstName string, lastName string) {
	if firstName != "" && lastName != "" {
		*relatives = append(*relatives, Relative{
			FirstName: firstName,
			LastName:  lastName,
			Relationship: relationship,
		})
	}
}